src/storage/copier/read_file_state.cpp

```diff
@@ -87,6 +87,56 @@ std::unique_ptr<ReadFileMorsel> ReadCSVSharedState::getMorselSerial() {
     return result;
 }
 
+// TODO(Guodong): Refactor duplicated between the three getMorsel() functions.
+std::unique_ptr<ReadFileMorsel> ReadCSVSharedState::getMorselSerial() {
+    std::unique_lock lck{mtx};
+    auto rows = 0;
+    while (true) {
+        if (currFileIdx >= filePaths.size()) {
+            // No more files to read.
+            return nullptr;
+        }
+        auto filePath = filePaths[currFileIdx];
+        if (!reader) {
+            reader = make_shared<BufferedCSVReader>(
+                filePath, &csvReaderConfig, tableSchema, memoryManager);
+        }
+        auto output = std::make_shared<DataChunk>(tableSchema->getNumProperties());
```

Comment:
Don't create data chunk for each call here. It's unnecessarily adding extra costs. Instead, the `resultSet` already have a data chunk initialized and prepared for you to use, pass it as function param.